src/v/cluster/controller_backend.cc

```diff
@@ -588,12 +588,23 @@ ss::future<std::error_code> controller_backend::update_partition_replica_set(
 }
 
 ss::future<> controller_backend::add_to_shard_table(
-  model::ntp ntp, raft::group_id raft_group, uint32_t shard) {
+  model::ntp ntp,
+  raft::group_id raft_group,
+  uint32_t shard,
+  model::revision_id revision) {
     // update shard_table: broadcast
+
     return _shard_table.invoke_on_all(
-      [ntp = std::move(ntp), raft_group, shard](shard_table& s) mutable {
-          s.insert(std::move(ntp), shard);
-          s.insert(raft_group, shard);
+      [self = _self, ntp = std::move(ntp), raft_group, shard, revision](
```

Comment:
Don’t we want to copy the ntp on every core.

Reply:
we actually do copy it, the whole lambda is copied on each core.

Reply:
Right. But the ntp was moved.